code/game/objects/items/stacks/sheets/glass.dm

```diff
@@ -67,6 +67,10 @@ GLOBAL_LIST_INIT(glass_recipes, list (
 
 /obj/item/stack/sheet/glass/attackby(obj/item/W, mob/user, params)
 	..()
+	if(istype(W, /obj/item/lightreplacer))
+		W.attackby(src, user)
+		return
+
```

Comment:
Does this have to be here? Can't you put this in the `attack_obj()` of the light replacer?